tensorboard/webapp/experiments/store/experiments_selectors_test.ts

```diff
@@ -49,32 +55,42 @@ describe('experiments selectors', () => {
     });
   });
 
-  describe('#getExperimentNames', () => {
+  describe('#getDashboardExperimentNames', () => {
     let state: State;
 
     beforeEach(() => {
       const foo = buildExperiment({id: 'foo', name: 'foo name'});
       const bar = buildExperiment({id: 'bar', name: 'bar name'});
 
-      state = buildStateFromExperimentsState({
-        data: {
-          experimentMap: {foo, bar},
-        },
+      state = buildMockState({
+        ...buildStateFromAppRoutingState(
+          buildAppRoutingState({
+            activeRoute: {
+              routeKind: RouteKind.COMPARE_EXPERIMENT,
+              params: {
+                experimentIds: 'exp1:foo,exp2:bar,exp3:baz',
+              },
+            },
+          })
+        ),
+        ...buildStateFromExperimentsState({
+          data: {
+            experimentMap: {foo, bar},
+          },
+        }),
       });
     });
 
     it('translates experiment ids to experiment names', () => {
-      expect(
-        selectors.getExperimentNames(['foo', 'bar', 'baz'])(state)
-      ).toEqual({
+      expect(selectors.getDashboardExperimentNames(state)).toEqual({
         foo: 'foo name',
         bar: 'bar name',
       });
     });
 
     it('returns an empty object when no experiments are provided', () => {
-      expect(selectors.getExperimentNames([])(state)).toEqual({});
-      expect(selectors.getExperimentNames(['abc', '123'])(state)).toEqual({});
+      state.app_routing!.activeRoute!.routeKind = RouteKind.EXPERIMENTS;
```

Comment:
Can you do something like:

```
state = {
  ...state,
  ...buildStateFromAppRoutingState(
    // Override the state here.
  ),
}
```

Referring to 'app_routing' in its bare form, as discussed in previous PR, is a little controversial.